### PR TITLE
feat(console): extract total number from headers for useSWR

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Extract "Total-Number" from response header to useSWR's `data`, if `page` and `page_size` is present in url.

Is unnable to get response header in useSWR directly, so we have to "break" data.

## Linear Issue Reference

LOG-1979

## Testing

Local tested.

### CODE REVIEW

File: packages/console/src/hooks/use-swr-fetcher.ts [hunk 1]
```diff
@@ -1,18 +1,35 @@
...
 import useApi, { RequestError } from './use-api';
 
 const useSwrFetcher = () => {
   const api = useApi({ hideErrorToast: true });
+  const { t } = useTranslation();
   const fetcher = useCallback<BareFetcher>(
     async (resource, init) => {
       try {
         const response = await api.get(resource, init);
+        const data = (await response.json()) as ArbitraryObject;
```
Comment (demonzoo) [inline packages/console/src/hooks/use-swr-fetcher.ts:16]: I added the dom.d.ts in PR #432 , after that PR is merged, you can use
```
const data = await response.json<ArbitraryObject>();
```

File: packages/console/src/pages/Users/index.tsx [hunk 2]
```diff
@@ -102,6 +107,18 @@ const Users = () => {
           ))}
         </tbody>
       </table>
+      {/* TODO: LOG-1977 pagination component */}
+      {data?.[1] && (
+        <div>
+          current page: {pageIndex}, total page: {Math.ceil(data[1] / pageSize)},{' '}
```
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:113]: What is the last `{' '}`?

File: packages/console/src/pages/Users/index.tsx [hunk 3]
```diff
@@ -102,6 +107,18 @@ const Users = () => {
...
+          <Button
+            title="admin_console.connectors.next"
+            onClick={() => {
+              setPageIndex((previous) => previous + 1);
+            }}
+          />
```
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:119]: Should hide this button when it's already the last page. Also, where's the `Previous` button?

File: packages/console/src/hooks/use-swr-fetcher.ts [hunk 4]
```diff
@@ -1,18 +1,35 @@
...
-        return await response.json();
+        if (typeof resource === 'string' && resource.includes('?')) {
+          const parameters = new URLSearchParams(resource.split('?')[1]);
+
+          if (parameters.get('page') && parameters.get('page_size')) {
+            const number = response.headers.get('Total-Number');
+
+            if (number) {
+              throw new Error(t('admin_console.errors.missing_total_number'));
+            }
```
Comment (demonzoo) [inline packages/console/src/hooks/use-swr-fetcher.ts:26]: if (number === undefined) ?

File: packages/console/src/pages/Users/index.tsx [hunk 5]
```diff
@@ -80,7 +85,7 @@ const Users = () => {
               </td>
             </TableLoading>
           )}
-          {data?.map(({ id, name, username }) => (
+          {data?.[0].map(({ id, name, username }) => (
```
Comment (demonzoo) [inline packages/console/src/pages/Users/index.tsx:88]: Better use destructuring
```
const [tableRows, totalPageCount] = data;
```

File: packages/console/src/pages/Users/index.tsx (hunk 3 above)
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx:119]: same as above

**Comment by wangsijie** [comment]
Let's hold this PR and wait for Pagination component.

File: packages/console/src/pages/Users/index.tsx (hunk 5 above)
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx]: updated

File: packages/console/src/hooks/use-swr-fetcher.ts (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/hooks/use-swr-fetcher.ts]: updated

File: packages/console/src/pages/Users/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/pages/Users/index.tsx]: Replaced with "Pagination"

File: packages/console/src/hooks/use-swr-fetcher.ts (hunk 1 above)
Comment (wangsijie) [reply packages/console/src/hooks/use-swr-fetcher.ts]: updated

**Comment by wangsijie** [comment]
Pagination component added, now ready for review.

**Review by demonzoo — APPROVED** [review]

**Pull request merged**